module/Finna/src/Finna/RecordDriver/SolrDefault.php

```diff
@@ -62,4 +62,21 @@ public function __construct(
         parent::__construct($mainConfig, $recordConfig, $searchSettings);
         $this->searchSettings = $searchSettings;
     }
+
+    /**
+     * Return the collection search ID for this record.
+     *
+     * @return string
+     */
+    public function getCollectionSearchId(): string
+    {
+        if ($this->mainConfig->Hierarchy->showFullHierarchyTree) {
+            $id = $this->getUniqueID();
+            $topId = $this->getHierarchyTopID()[0] ?? '';
+            if ($topId && $topId !== $id) {
```

Comment:
Tän voi varmaan tiivistää; return $this->getHierarchyTopID()[0] ?? $this->getUniqueID(); ?